src/otx/core/data/transform_libs/torchvision.py

```diff
@@ -503,41 +517,74 @@ def __init__(
-            self.scale = None
-        elif isinstance(scale, int):
-            self.scale = (scale, scale)
+        if resize_size is not None and not (
+            isinstance(resize_size, int)
+            or (isinstance(resize_size, Sequence) and all(isinstance(r, int) for r in resize_size))
+        ):
+            msg = f"resize_size must be int or Sequence[int], but got `{type(resize_size)}`."
+            raise TypeError(msg)
+
+        if scale_factor is None:
+            self.scale_factor = None
+        elif isinstance(scale_factor, float):
+            self.scale_factor = (scale_factor, scale_factor)
+        elif isinstance(scale_factor, Sequence) and not all(isinstance(sf, float) for sf in scale_factor):
+            msg = (
+                "scale_factor must be float or Sequence[float], "
+                f"but got `{type(scale_factor)}[{type(scale_factor[0])}]`."
+            )
+            raise TypeError(msg)
+        elif isinstance(scale_factor, Sequence) and len(scale_factor) == 2:
+            self.scale_factor = tuple(scale_factor)
         else:
-            self.scale = tuple(scale)  # type: ignore[assignment]
+            msg = f"expect scale_factor is float or Sequence[float], but get {type(scale_factor)}"
+            raise TypeError(msg)
 
+        self.input_size = (input_size, input_size) if isinstance(input_size, int) else input_size
         self.transform_bbox = transform_bbox
         self.transform_point = transform_point
         self.transform_mask = transform_mask
         self.interpolation = interpolation
         self.interpolation_mask = interpolation_mask
         self.keep_ratio = keep_ratio
         self.clip_object_border = clip_object_border
-        if scale_factor is None:
-            self.scale_factor = None
-        elif isinstance(scale_factor, float):
-            self.scale_factor = (scale_factor, scale_factor)
-        elif isinstance(scale_factor, tuple) and len(scale_factor) == 2:
-            self.scale_factor = scale_factor
-        else:
-            msg = f"expect scale_factor is float or Tuple(float), butget {type(scale_factor)}"
-            raise TypeError(msg)
-
         self.is_numpy_to_tvtensor = is_numpy_to_tvtensor
 
+        # for configurable input size
+        self._resize_size = resize_size
+        self.resize_size_scale = resize_size_scale
+
+    @property
+    def resize_size(self) -> tuple[int, int] | None:
+        """Resize size for resizing with (height, width)."""
+        if self.scale_factor is not None:
+            return None
+
+        if self._resize_size is None:
+            self._resize_size = (
+                int(self.input_size[0] * self.resize_size_scale[0]),
+                int(self.input_size[1] * self.resize_size_scale[1]),
+            )
+        elif isinstance(self._resize_size, int):
+            self._resize_size = (self._resize_size, self._resize_size)
+
+        return (self._resize_size[0], self._resize_size[1])
```

Comment:
```suggestion
        return self._resize_size
```
If self._resize_size is `tuple[int, int]`, how about retuninig it?